app/src/main/java/com/github/se/cyrcle/ui/gambling/GamblingScreen.kt

```diff
@@ -245,21 +249,52 @@ fun WheelView(
 }
 
 @Composable
-fun GamblingScreen() {
+fun GamblingScreen(navigationActions: NavigationActions, userViewModel: UserViewModel) {
+  val userState by userViewModel.currentUser.collectAsState()
+  val coins = userState?.details?.wallet?.getCoins()
+  val spinCost = 10 // Cost to spin the wheel
+
   Box(
       modifier = Modifier.fillMaxSize().testTag("gambling_screen"),
       contentAlignment = Alignment.Center) {
         var wheelSpinFunction by remember { mutableStateOf<(() -> Unit)?>(null) }
 
+        Text(
+            text = "Coins: ${coins ?: "BROKIE"}",
+            modifier =
+                Modifier.align(Alignment.TopCenter).padding(top = 16.dp).testTag("coin_display"),
+            style = MaterialTheme.typography.headlineMedium)
+
         WheelView(
                 modifier = Modifier.size(300.dp).testTag("wheel_view"),
                 onSegmentLanded = { /* Handle landing */})
             .let { spinFn -> wheelSpinFunction = spinFn }
 
+        val canSpin = userState?.details?.wallet?.isSolvable(spinCost, -1) == true
+        // I really dont like the fact that I have to put -1 as threshold,
+        // but it is this way because of how André wrote the isSolvable function.
+        // If i want to put 0 here then technically all that I have to do is change the > into >=
```

Comment:
about your comment you totally can change André's code, he did the groundwork but if it turns out it's not ideal, it can be changed since the function is not used anywhere yet

Reply:
Ok ill definitely change that then.